src/pages/home/report/ReportActionCompose/SuggestionEmoji.js

```diff
@@ -0,0 +1,267 @@
+import React, {useState, useCallback, useRef, useImperativeHandle} from 'react';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import {withOnyx} from 'react-native-onyx';
+import CONST from '../../../../CONST';
+import useArrowKeyFocusManager from '../../../../hooks/useArrowKeyFocusManager';
+import * as SuggestionsUtils from '../../../../libs/SuggestionUtils';
+import * as EmojiUtils from '../../../../libs/EmojiUtils';
+import EmojiSuggestions from '../../../../components/EmojiSuggestions';
+import ONYXKEYS from '../../../../ONYXKEYS';
+import useLocalize from '../../../../hooks/useLocalize';
+import * as SuggestionProps from './suggestionProps';
+
+/**
+ * Check if this piece of string looks like an emoji
+ * @param {String} str
+ * @param {Number} pos
+ * @returns {Boolean}
+ */
+const isEmojiCode = (str, pos) => {
+    const leftWords = str.slice(0, pos).split(CONST.REGEX.SPECIAL_CHAR_OR_EMOJI);
+    const leftWord = _.last(leftWords);
+    return CONST.REGEX.HAS_COLON_ONLY_AT_THE_BEGINNING.test(leftWord) && leftWord.length > 2;
+};
+
+const defaultSuggestionsValues = {
+    suggestedEmojis: [],
+    colonSignIndex: -1,
+    shouldShowSuggestionMenu: false,
+};
+
+const propTypes = {
+    /** Preferred skin tone */
+    preferredSkinTone: PropTypes.number,
+
+    /** A ref to this component */
+    forwardedRef: PropTypes.shape({current: PropTypes.shape({})}),
+
+    /** Function to clear the input */
+    resetKeyboardInput: PropTypes.func.isRequired,
+
+    /** Callback when a emoji was inserted */
+    onInsertedEmoji: PropTypes.func.isRequired,
+
+    /** The current selection */
+    selection: PropTypes.shape({
+        start: PropTypes.number.isRequired,
+        end: PropTypes.number.isRequired,
+    }).isRequired,
+
+    ...SuggestionProps.baseProps,
+};
+
+const defaultProps = {
+    preferredSkinTone: CONST.EMOJI_DEFAULT_SKIN_TONE,
+    forwardedRef: null,
+};
+
+function SuggestionEmoji({
+    preferredSkinTone,
+    value,
+    setValue,
+    selection,
+    setSelection,
+    updateComment,
+    isComposerFullSize,
+    shouldShowReportRecipientLocalTime,
+    isAutoSuggestionPickerLarge,
+    forwardedRef,
+    resetKeyboardInput,
+    onInsertedEmoji,
+    measureParentContainer,
+}) {
+    const [suggestionValues, setSuggestionValues] = useState(defaultSuggestionsValues);
+
+    const isEmojiSuggestionsMenuVisible = !_.isEmpty(suggestionValues.suggestedEmojis) && suggestionValues.shouldShowEmojiSuggestionMenu;
+
+    const [highlightedEmojiIndex, setHighlightedEmojiIndex] = useArrowKeyFocusManager({
+        isActive: isEmojiSuggestionsMenuVisible,
+        maxIndex: SuggestionsUtils.getMaxArrowIndex(suggestionValues.suggestedEmojis.length, isAutoSuggestionPickerLarge),
```

Comment:
👋 Coming from https://github.com/Expensify/App/issues/33613
`SuggestionsUtils.getMaxArrowIndex` did return visible list length (instead of true list length), which meant that you couldn't scroll past the visible items using a keyboard
We resolved this with
`maxIndex: suggestionValues.suggestedMentions.length - 1`